sdk/Sdk.Generators/ExtensionStartupRunnerGenerator.cs

```diff
@@ -53,21 +54,43 @@ public class ExtensionStartupRunnerGenerator : ISourceGenerator
         /// <summary>
         /// Fully qualified name of the above "WorkerExtensionStartupAttribute" attribute.
         /// </summary>
-        private const string AttributeTypeFullName = "Microsoft.Azure.Functions.Worker.Core.WorkerExtensionStartupAttribute";
+        private const string AttributeTypeFullName =
+            "Microsoft.Azure.Functions.Worker.Core.WorkerExtensionStartupAttribute";
 
         /// <summary>
         /// Fully qualified name of the base type which extension startup classes should implement.
         /// </summary>
         private const string StartupBaseClassName = "Microsoft.Azure.Functions.Worker.Core.WorkerExtensionStartup";
 
-        public void Execute(GeneratorExecutionContext context)
+        public void Initialize(IncrementalGeneratorInitializationContext context)
         {
-            var extensionStartupTypeNames = GetExtensionStartupTypes(context);
+            // Get the sub set of assemblies with startup attribute
+            var assembliesValueProvider = context.CompilationProvider
+                .SelectMany((compilation, token) =>
```

Comment:
I see examples of incremental generators declaring their lambdas with `static`. Is that something we should be doing as well? From my understanding, it will enforce no capturing of execution context (so a perf safeguard).

Reply:
Done.